src/serviceregistry/manual/examples.md

```diff
@@ -0,0 +1,2390 @@
+
+# Discovery endpoints
+
+## **Device discovery**
+- **Register device:** POST /serviceregistry/device-discovery/register
+  
+  Request body: 
+  ~~~
+  {
+    "name": "thermometer1",
+    "metadata": {
+      "scales": ["Kelvin", "Celsius"],
+      "max-temperature": {"Kelvin": 310, "Celsius": 40},
+      "min-temperature": {"Kelvin": 260, "Celsius": -10},
+      "appliable": ["spring", "autumn"]
+    },
+    "addresses": [
+      {
+        "type": "MAC",
+        "address": "81:ef:1a:44:7a:ff"
+      }
+    ]
+  }
+  ~~~
+
+  Response body:
+  ~~~
+  {
+    "name": "thermometer1",
+    "metadata": {
+      "scales": [
+        "Kelvin",
+        "Celsius"
+      ],
+      "max-temperature": {
+        "Kelvin": 310,
+        "Celsius": 40
+      },
+      "min-temperature": {
+        "Kelvin": 260,
+        "Celsius": -10
+      },
+      "appliable": [
+        "spring",
+        "autumn"
+      ]
+    },
+    "addresses": [
+      {
+        "type": "MAC",
+        "address": "81:ef:1a:44:7a:ff"
+      }
+    ],
+    "createdAt": "2024-10-21T18:47:45.388404800Z",
+    "updatedAt": "2024-10-21T18:47:45.388404800Z"
+  }
+- **Lookup device:** POST /serviceregistry/device-discovery/lookup
+  
+  Request body:
+  ~~~
+  {
+    "deviceNames": [
+    ],
+    "addresses": [
+      "81:ef:1a:44:7a:ff", "ab:ff:c8:e1:45:9a"
+    ],
+    "addressType": "MAC",
+    "metadataRequirementList": [
+    ]
+  }
+  ~~~
+
+  Response body:
+  ~~~
+  {
+    "entries": [
+      {
+        "name": "thermometer1",
+        "metadata": {
+          "scales": [
+            "Kelvin",
+            "Celsius"
+          ],
+          "max-temperature": {
+            "Kelvin": 310,
+            "Celsius": 40
+          },
+          "min-temperature": {
+            "Kelvin": 260,
+            "Celsius": -10
+          },
+          "appliable": [
+            "spring",
+            "autumn"
+          ]
+        },
+        "addresses": [
+          {
+            "type": "MAC",
+            "address": "81:ef:1a:44:7a:ff"
+          }
+        ],
+        "createdAt": "2024-10-21T18:47:45Z",
+        "updatedAt": "2024-10-21T18:47:45Z"
+      },
+      {
+        "name": "weather-displayer1",
+        "metadata": {
+          "type": "analogue",
+          "displayed-data": [
+            "temperature"
+          ]
+        },
+        "addresses": [
+          {
+            "type": "MAC",
+            "address": "ab:ff:c8:e1:45:9a"
+          }
+        ],
+        "createdAt": "2024-10-22T09:44:20Z",
+        "updatedAt": "2024-10-22T09:44:20Z"
+      }
+    ],
+    "count": 2
+  }
+  ~~~
+
+- **Revoke device:** DELETE /serviceregistry/device-discovery/revoke/{name}
+  
+  Query parameter: test-weather-displayer
+  ~~~
+  http://localhost:8443/serviceregistry/device-discovery/revoke/test-weather-displayer
+  ~~~
+
+  Response code: 200 
+
+##  **System discovery**
+
+- **Register system:** POST /serviceregistry/system-discovery/register
```

Comment:
should mention that system name comes from Authorization header or X.509 certificate. Inject an example with self-declared authentication.